src/skydb.ts

```diff
@@ -1,105 +1,79 @@
-import { md, pkcs5, pki } from "node-forge";
+import { pki } from "node-forge";
 import { SkynetClient } from "./client";
-import { HashFileID, HashRegistryValue } from "./crypto";
-import { RegistryValue, SignedRegistryValue } from "./registry";
-import { promiseTimeout, trimUriPrefix, uriSkynetPrefix } from "./utils";
+import { HashRegistryEntry, PublicKey, SecretKey } from "./crypto";
+import { RegistryEntry } from "./registry";
+import { parseSkylink, trimUriPrefix, uriSkynetPrefix } from "./utils";
 
-// FILEID_V1 represents version 1 of the FileID object
-export const FILEID_V1 = 1;
-
-// FileType is the type of the file
-export enum FileType {
-  Invalid, // 0 is invalid
-  PublicUnencrypted,
-}
-
-// getFile will lookup the entry for given skappID and filename, if it exists it
-// will try and download the file behind the skylink it has found in the entry.
-export async function getFile(this: SkynetClient, user: User, fileID: FileID): Promise<SkyFile> {
+export async function getJSON(this: SkynetClient, publicKey: PublicKey, dataKey: string): Promise<{ json: object, revision: number }|null> {
   // lookup the registry entry
-  const existing = await this.lookupRegistry(user, fileID);
-  if (!existing) {
+  const entry = await this.registry.lookup(publicKey, dataKey);
+  if (entry === null) {
     throw new Error("not found");
   }
 
-  // download the data in that Skylink
-  const skylink = existing.value.data;
+  // Download the data in that Skylink
+  // TODO: Replace with download request method.
+  let skylink: string;
+  try {
+    skylink = parseSkylink(entry.value.data);
+  } catch (error) {
+    throw new Error(`invalid skylink: ${error}`);
+  }
+
   const response = await this.executeRequest({
     ...this.customOptions,
     method: "get",
     url: this.getSkylinkUrl(skylink),
   });
 
-  // wrap the data in a skyfile and return it
-  const metadata = JSON.parse(response.headers["skynet-file-metadata"]);
-  const file = new SkyFile(new File([response.data], metadata.filename, { type: response.headers["content-type"] }));
-  return file;
+  return { json: response.data, revision: entry.revision };
```

Comment:
Let's just call it `data` so it's more generic and we could use the same response type in future `getText` or `getBuffer` methods. We already know it's json because it's the response from `getJson`. Actually, it's not even json - it's json parsed as an object :)

```suggestion
  return { data: response.data, revision: entry.revision };
```